src/rust/storage/seg/src/tests.rs

```diff
@@ -0,0 +1,337 @@
+use super::*;
+use crate::hashtable::HashBucket;
+use crate::item::ITEM_HDR_SIZE;
+use core::num::NonZeroU32;
+
+#[test]
+fn sizes() {
+    #[cfg(feature = "magic")]
+    assert_eq!(ITEM_HDR_SIZE, 9);
+
+    #[cfg(not(feature = "magic"))]
+    assert_eq!(ITEM_HDR_SIZE, 5);
+
+    assert_eq!(std::mem::size_of::<Segments>(), 64);
+    assert_eq!(std::mem::size_of::<SegmentHeader>(), 64);
+
+    assert_eq!(std::mem::size_of::<HashBucket>(), 64);
+    assert_eq!(std::mem::size_of::<HashTable>(), 64);
+
+    assert_eq!(std::mem::size_of::<crate::ttl_buckets::TtlBucket>(), 64);
+    assert_eq!(std::mem::size_of::<TtlBuckets>(), 16);
+}
+
+#[test]
+fn init() {
+    let mut cache = SegCache::builder()
```

Comment:
`cargo test` doesn't build. `SegCache` needs to be updated to `Seg` throughout this file.